### PR TITLE
8333729: C2 SuperWord: remove some @requires usages in test/hotspot/jtreg/compiler/loopopts/superword

### PR DESCRIPTION
**Pull request by eme64**
I'm removing as many `@requires` statements as possible. Generally, they just limit what tests can be executed where, and in most cases such limitations do not make much sense.

For example, tests should not be limited to `@requires vm.compiler2.enabled`, i.e. only run if C2 is available - we can also run the tests if GraalVM is available.

We should also not limit tests to `@requires vm.bits == "64"`, but rather limit the `@IR` rules with `applyIfPlatform = {"64-bit", "true"}`.

Similarly, we should not limit the platform for a run statement, but do that per `@IR` rule if necessary.

Update: I had to add some extra `-XX:+IgnoreUnrecognizedVMOptions`, so that the C2 flags do not make the VM crash if C2 is not available. I ran all the relevant tests on a build without C2, see discussion in the thread below.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8333729](https://bugs.openjdk.org/browse/JDK-8333729): C2 SuperWord: remove some @requires usages in test/hotspot/jtreg/compiler/loopopts/superword (**Sub-task** - P4)

### Reviewers
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**) ⚠️ Review applies to [2733a868](https://git.openjdk.org/jdk/pull/19581/files/2733a8689b27087b68e9f22bd8223989c2a31cac)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19581/head:pull/19581` \
`$ git checkout pull/19581`

Update a local copy of the PR: \
`$ git checkout pull/19581` \
`$ git pull https://git.openjdk.org/jdk.git pull/19581/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19581`

View PR using the GUI difftool: \
`$ git pr show -t 19581`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19581.diff">https://git.openjdk.org/jdk/pull/19581.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19581#issuecomment-2162424484)

### CODE REVIEW

**Review by chhagedorn — APPROVED** [review]
That's a good idea to remove these limitations. Changes look good.

File: test/hotspot/jtreg/compiler/loopopts/superword/CoLocatePackMemoryState.java [hunk 1]
```diff
@@ -24,7 +24,6 @@
 
 /**
  * @test
- * @requires vm.compiler2.enabled
```
Comment (vnkozlov) [inline test/hotspot/jtreg/compiler/loopopts/superword/CoLocatePackMemoryState.java:27]: Some tests uses this requires when they use C2 specific flags. Like in this case: `LoopMaxUnroll` is C2 flag.
If you want to run such tests with any VM (which may not include C2) you need to add `-XX:+IgnoreUnrecognizedVMOptions`

**Review by vnkozlov — COMMENTED** [review]
I suggest to build VM without C2 and run these modified tests with to find missing `-XX:+IgnoreUnrecognizedVMOptions` flag and other failure modes.

Also build and run 32-bit VM with these tests too.

**Comment by eme64** [comment]
@vnkozlov Ok, I can do that. Would be nice if we had such testing automatically in our testing though - build and test without C2.

**Comment by eme64** [comment]
What I am doing now, work in progress:
```
bash jib.sh configure -- --with-debug-level=fastdebug --with-conf-name=linux-x64-debug-noC2 --with-jvm-features=-compiler2
make hotspot CONF=linux-x64-debug-noC2; make jdk CONF=linux-x64-debug-noC2;
```
And running all relevant tests:
```
~/Documents/jtreg/bin/jtreg -va -s -jdk:/oracle-work/jdk-fork2/build/linux-x64-debug-noC2/jdk -javaoptions:"" -J-Djavatest.maxOutputSize=10000000 /oracle-work/jdk-fork2/open/test/hotspot/jtreg/compiler/loopopts/superword/
```

And then I can add `-XX:+IgnoreUnrecognizedVMOptions` in cases where we are using C2 flags, just as  @vnkozlov suggested.

**Comment by eme64** [comment]
```
[compiler/loopopts/superword/ReductionPerf.java](https://github.com/openjdk/JTwork/compiler/loopopts/superword/ReductionPerf.jtr): Test SuperWord Reduction Perf.
[compiler/loopopts/superword/TestAlignVector.java#AlignVector](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestAlignVector_AlignVector.jtr): Test AlignVector with various loop init, stride, scale, invar, etc.
[compiler/loopopts/superword/TestAlignVector.java#NoAlignVector](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestAlignVector_NoAlignVector.jtr): Test AlignVector with various loop init, stride, scale, invar, etc.
[compiler/loopopts/superword/TestDependencyOffsets.java#vanilla-A](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestDependencyOffsets_vanilla-A.jtr): Test SuperWord: vector size, offsets, dependencies, alignment.
[compiler/loopopts/superword/TestDependencyOffsets.java#vanilla-U](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestDependencyOffsets_vanilla-U.jtr): Test SuperWord: vector size, offsets, dependencies, alignment.
[compiler/loopopts/superword/TestGeneralizedReductions.java](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestGeneralizedReductions.jtr): Test reduction vectorizations that are enabled by performing SLP reduction analysis on unrolled loops.
[compiler/loopopts/superword/TestIndependentPacksWithCyclicDependency.java](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestIndependentPacksWithCyclicDependency.jtr): Test some examples with independent packs with cyclic dependency between the packs.
[compiler/loopopts/superword/TestIndependentPacksWithCyclicDependency2.java](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestIndependentPacksWithCyclicDependency2.jtr): Test some examples with independent packs with cyclic dependency between the packs.
[compiler/loopopts/superword/TestLargeCompilation.java](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestLargeCompilation.jtr): Test compile time for large compilation, where SuperWord takes especially much time.
[compiler/loopopts/superword/TestLargeScaleAndStride.java#AlignVector](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestLargeScaleAndStride_AlignVector.jtr):
[compiler/loopopts/superword/TestMovingLoadBeforeStore.java](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestMovingLoadBeforeStore.jtr): In SuperWord::output, LoadVector can be moved before StoreVector, but only if it is proven to be safe.
[compiler/loopopts/superword/TestMulAddS2I.java](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestMulAddS2I.jtr): Test MulAddS2I vectorization.
[compiler/loopopts/superword/TestPeeledReductionNode.java](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestPeeledReductionNode.jtr): Test that reduction nodes peeled out of an inner loop are not vectorized as reductions within the outer loop.
[compiler/loopopts/superword/TestScheduleReordersScalarMemops.java](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestScheduleReordersScalarMemops.jtr): Test some examples where non-vectorized memops also need to be reordered during SuperWord::schedule.
[compiler/loopopts/superword/TestSplitPacks.java](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestSplitPacks.jtr): Test splitting packs in SuperWord
[compiler/loopopts/superword/TestUnorderedReduction.java#MaxVectorSize16-Unaligned](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestUnorderedReduction_MaxVectorSize16-Unaligned.jtr): Special test cases for PhaseIdealLoop::move_unordered_reduction_out_of_loop
[compiler/loopopts/superword/TestUnorderedReduction.java#MaxVectorSize32-Aligned](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestUnorderedReduction_MaxVectorSize32-Aligned.jtr): Special test cases for PhaseIdealLoop::move_unordered_reduction_out_of_loop
[compiler/loopopts/superword/TestUnorderedReduction.java#Vanilla-Aligned](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestUnorderedReduction_Vanilla-Aligned.jtr): Special test cases for PhaseIdealLoop::move_unordered_reduction_out_of_loop
[compiler/loopopts/superword/TestUnorderedReduction.java#Vanilla-Unaligned](https://github.com/openjdk/JTwork/compiler/loopopts/superword/TestUnorderedReduction_Vanilla-Unaligned.jtr): Special test cases for PhaseIdealLoop::move_unordered_reduction_out_of_loop
[compiler/loopopts/superword/Vec_MulAddS2I.java](https://github.com/openjdk/JTwork/compiler/loopopts/superword/Vec_MulAddS2I.jtr): Test operations in C2 MulAddS2I and MulAddVS2VI nodes.
```
All of these now are failing without C2, investigating...

**Comment by eme64** [comment]
@vnkozlov thanks for that catch. I now fixed all occurrences. I tested it again on a build without C2, and all tests pass now :)

**Review by vnkozlov — APPROVED** [review]
Looks good now.

**Comment by eme64** [comment]
Thanks @vnkozlov @chhagedorn for the reviews!
/integrate

**Pull request closed without merging**